packages/babel-core/src/transformation/plugin.js

```diff
@@ -77,6 +77,10 @@ export default class Plugin extends Store {
 
     this.maybeInherit(loc);
 
+    if (this.raw._c) {
+      throw new Error(`Plugin ${JSON.stringify(i)} specified ${JSON.stringify(loc)} appears to be a different plugin instance. Possible duplicate babel-core versions.`);
```

Comment:
No idea. I just found this when debugging an issue with the Relay plugin, Babel was throwing a super cryptic message.